File: CS/MVVMDemo.Wpf/Modules/DXCommand/BindingCommandToMethodsViewModel.cs
using DevExpress.Mvvm;
using System.Windows;

namespace MVVMDemo.DXCommandDemo {
    public class BindingCommandToMethodsViewModel : BindableBase {
        public bool CanExecuteSaveCommand {
            get { return GetProperty(() => CanExecuteSaveCommand); }
            set { SetProperty(() => CanExecuteSaveCommand, value); }
        }
        public void Save() {
            MessageBox.Show("Action: Save");
        }

        public string FileName {
            get { return GetProperty(() => FileName); }
            set { SetProperty(() => FileName, value); }
        }
        public void Open(string fileName) {
            MessageBox.Show(string.Format("Action: Open {0}", fileName));
        }

        public BindingCommandToMethodsViewModel() {
            CanExecuteSaveCommand = true;
        }

    }
}

File: VB/MVVMDemo.Wpf/Modules/DXCommand/BindingCommandToMethodsViewModel.vb
Imports DevExpress.Mvvm
Imports System.Windows

Namespace MVVMDemo.DXCommandDemo
    Public Class BindingCommandToMethodsViewModel
        Inherits BindableBase

        Public Property CanExecuteSaveCommand() As Boolean
            Get
                Return GetProperty(Function() CanExecuteSaveCommand)
            End Get
            Set(ByVal value As Boolean)
                SetProperty(Function() CanExecuteSaveCommand, value)
            End Set
        End Property
        Public Sub Save()
            MessageBox.Show("Action: Save")
        End Sub

        Public Property FileName() As String
            Get
                Return GetProperty(Function() FileName)
            End Get
            Set(ByVal value As String)
                SetProperty(Function() FileName, value)
            End Set
        End Property
        Public Sub Open(ByVal fileName As String)
            MessageBox.Show(String.Format("Action: Open {0}", fileName))
        End Sub

        Public Sub New()
            CanExecuteSaveCommand = True
        End Sub

    End Class
End Namespace
